virtme/guest/virtme-sshd-script

```diff
@@ -46,4 +46,11 @@ for key in "${CACHE_DIR}"/etc/ssh/ssh_host_*_key; do
     ARGS+=(-h "${key}")
 done
 
-/usr/sbin/sshd "${ARGS[@]}"
+if [ ! -c /dev/vsock ]; then
+    /usr/sbin/sshd "${ARGS[@]}"
+else
+    # 4294967295 == U32_MAX == -1
+    declare -r VMADDR_CID_ANY=4294967295
+    # TODO Use something like syslog or journal for the logging
+    nohup systemd-socket-activate --accept --listen="vsock:${VMADDR_CID_ANY}:22" --inetd -- /usr/sbin/sshd -i "${ARGS[@]}" &> /dev/null &
```

Comment:
(maybe better to use this notation because `nohup` will do more than needed and be a bit chatty) `(systemd-socket-activate &)`)

Reply:
I've just checked the `sshd` source code what they do to become a daemon [1], because that's what did before when we've called `/usr/sbin/sshd ...` [2]. 

1. fork
2. call `setsid`
3. redirect stdin, stdout and stderr FDs to /dev/null

So if you want, we can do the same and use `setsid` instead of `nohup`.

[1] https://github.com/openssh/openssh-portable/blob/a73890e340fbd6121251854b658a72d738b86c84/openbsd-compat/daemon.c#L52
[2] https://github.com/arighi/virtme-ng/blob/main/virtme/guest/virtme-sshd-script#L49

Reply:
Looks good now, thanks @mhartmay and @matttbe !

Reply:
Hmm need to test something, seems I broke SSH over vsock + Rust init.

Reply:
I've added a "hack". See https://github.com/arighi/virtme-ng/pull/261/files#diff-ca70a7ba8e943bcf273078a8e5ed5314fa77bf70a93271a2da519e19efe9af5cR51.

Reply:
I've improved this a little and should now work with kernel module and kernel builtin vsock support. [1]
It also gives an error

https://github.com/arighi/virtme-ng/pull/261/commits/0027dbfb6af208782e89b0c943f1c4c9ec123782#diff-ca70a7ba8e943bcf273078a8e5ed5314fa77bf70a93271a2da519e19efe9af5cR56

if the support in the guest kernel is missing.